builtin/credential/aws/path_role_test.go

```diff
@@ -986,6 +988,87 @@ func TestAwsVersion(t *testing.T) {
 	}
 }
 
+// This test was used to reproduce https://github.com/hashicorp/vault/issues/7418
+// and verify its fix.
+// Please run it at least 3 times to ensure that passing tests are due to actually
+// passing, rather than the region being randomly chosen tying to the one in the
+// test through luck.
+func TestRoleResolutionWithSTSEndpointConfigured(t *testing.T) {
+	if enabled := os.Getenv("VAULT_ACC"); enabled == "" {
+		t.Skip()
+	}
+
+	// ex. "arn:aws:iam::123456789012:role/MyRole"
+	assumableRoleArn := os.Getenv("AWS_ASSUMABLE_ROLE_ARN")
+	if assumableRoleArn == "" {
+		t.Skip("skipping because AWS_ASSUMABLE_ROLE_ARN is unset")
+	}
+
+	// Ensure aws credentials are available locally for testing.
```

Comment:
`TestBackendAcc_LoginWithInstanceIdentityDocAndWhitelistIdentity` and `TestBackendAcc_LoginWithCallerIdentity` will actually override the `AWS_*` environment variables with `TEST_AWS_*` -- this creates a little bit of inconsistency. Maybe extract that overriding logic into a common function (e.g., `setupTestCreds`) to also be called here?

Reply:
Hm, that has the potential to expand the scope of this PR to tuning tests. For instance, I'm looking at `TestBackendAcc_LoginWithInstanceIdentityDocAndWhitelistIdentity` and I'm noticing that [here](https://github.com/hashicorp/vault/blob/805a0bc9b4605aec2b1bdf5aa1cff5cc43dc78f6/builtin/credential/aws/backend_test.go#L1093), the code verifies that the env vars aren't empty and subsequently uses them, but [here](https://github.com/hashicorp/vault/blob/805a0bc9b4605aec2b1bdf5aa1cff5cc43dc78f6/builtin/credential/aws/backend_test.go#L1099) it checks for an impossible condition and fails if it's found. I'm thinking it might make more sense to leave this test as-is because it's consistent with code already in Vault, and to make a separate ticket to circle back and resolve test inconsistencies.